test/gtest/uct/uct_test.cc

```diff
@@ -801,7 +828,13 @@ void uct_test::entity::connect(unsigned index, entity& other,
 
 void uct_test::entity::connect(unsigned index, entity& other, unsigned other_index)
 {
-    connect(index, other, other_index, NULL);
+    if (iface_attr().cap.flags & UCT_IFACE_FLAG_CONNECT_TO_EP) {
```

Comment:
why are you splitting this? IMHO it's better to cover all the connect modes in one function..

Reply:
IMO, better to use C++ overloading mechanism and don't pass raw pointers

Reply:
respectfully disagree :)